### PR TITLE
Refactor testutil, ModuleListTest, ModelManagerTest and add necessary classes for Module testing

### LINKED ISSUE #203: Test: Refactor Testutil package
Note :
- Refactor after the code in the main package is settled.
- CommandTestUtil also needs to be refactored since TypicalPersons.java depends on it.

### PR DESCRIPTION
**Pull request by michael-setia**
Fixes #203 

Refactored ModelManagerTest and ModuleList and added some necessary files for Module testing.

File changes: 
- CommandTestUtil class :
  - Added static field VALID_MODULENAME_CS2103T 
  - Added static field VALID_ZOOMLINK_CS2103T
  - Added static field VALID_MODULENAME_ES2660 
  - Added static field VALID_ZOOMLINK_ES2660

- ModelManagerTest class :
  - Has been refactored
  - All the test, except equals() method, can now function properly

- ModuleListTest class : 
  - Has been refactored
  - All test, except resetData_withDuplicateModules_throwsDuplicateModuleException() method, can now function properly

- AddressBookBuilder class : 
  - Renamed to -> ModuleListBuilder
  - Has been refactored

Added new files : 
- EditModuleDescriptorBuilder class
- TypicalModules class

Note : the changes do not cover everything since some of the necessary classes for the test have not been created, etc.

### CODE REVIEW

File: src/test/java/seedu/address/model/ModelManagerTest.java [hunk 1]
```diff
@@ -113,16 +112,18 @@ public void equals() {
         // different types -> returns false
         assertFalse(modelManager.equals(5));
 
-        // different addressBook -> returns false
-        // assertFalse(modelManager.equals(new ModelManager(differentModuleList, userPrefs)));
+        // different moduleList -> returns false
+        assertFalse(modelManager.equals(new ModelManager(differentModuleList, userPrefs)));
+
+        // Note : No NameContainsKeywordsPredicate class for Module yet.
```
Comment (murtubak) [inline src/test/java/seedu/address/model/ModelManagerTest.java:118]: We should discuss what predicates to use for this class

File: src/test/java/seedu/address/model/ModuleListTest.java [hunk 2]
```diff
@@ -36,65 +37,65 @@ public void resetData_null_throwsNullPointerException() {
...
+    public void hasModule_moduleInModuleList_returnsTrue() {
+        moduleList.addModule(CS2030);
+        assertTrue(moduleList.hasModule(CS2030));
     }
 
     @Test
-    public void hasPerson_personWithSameIdentityFieldsInAddressBook_returnsTrue() {
-        // moduleList.addPerson(ALICE);
-        Person editedAlice = new PersonBuilder(ALICE).withTags(VALID_TAG_HUSBAND)
+    public void hasModule_moduleWithSameNameModuleList_returnsTrue() {
```
Comment (murtubak) [inline src/test/java/seedu/address/model/ModuleListTest.java:74]: This test is not updated for additional info added to Module like Grade Tracker which I recently added. I will update this in the future.

File: src/test/java/seedu/address/testutil/EditModuleDescriptorBuilder.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    public EditModuleDescriptorBuilder() {
+        descriptor = new EditModuleDescriptor();
+    }
+
+    public EditModuleDescriptorBuilder(EditModuleDescriptor descriptor) {
+        descriptor = new EditModuleDescriptor(descriptor);
+    }
+
+    /**
+     * Returns an {@code EditModuleDescriptor} with fields containing {@code module}'s details
```
Comment (murtubak) [inline src/test/java/seedu/address/testutil/EditModuleDescriptorBuilder.java:24]: Also not updated for new info in the Module class, will update in the future commits

File: src/test/java/seedu/address/testutil/ModuleBuilder.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    public static final String DEFAULT_MODULENAME = "CS2103T";
+    public static final String DEFAULT_ZOOMLINK = "www.zoom.us";
+
+    private ModuleName moduleName;
+    private ZoomLink zoomLink;
+
+    /**
+     * Creates a {@code ModuleBuilder} with the default details.
+     */
+    public ModuleBuilder() {
```
Comment (murtubak) [inline src/test/java/seedu/address/testutil/ModuleBuilder.java:21]: Also needs updating for additional info put into module.

**Review by murtubak — APPROVED** [review]
LGTM. Future changes to module will  require us to review some of these tests

File: src/test/java/seedu/address/model/ModelManagerTest.java (hunk 1 above)
Comment (michael-setia) [reply src/test/java/seedu/address/model/ModelManagerTest.java:118]: Okay, sure.

File: src/test/java/seedu/address/model/ModuleListTest.java (hunk 2 above)
Comment (michael-setia) [reply src/test/java/seedu/address/model/ModuleListTest.java:74]: Yes, sure. If the test doesn't work, we can always comment out first then make the changes later.

**Pull request merged**